### ISSUE TITLE
Error installing on macOS Sierra

### ISSUE THREAD

**Issue by heathermiller** [inukshuk/jekyll-scholar#171]
I'm no Ruby expert, but I'm struggling to install on macOS Sierra:
```
$ sudo gem install jekyll-scholar
Password:
Fetching: namae-0.11.0.gem (100%)
Successfully installed namae-0.11.0
Fetching: citeproc-1.0.4.gem (100%)
Successfully installed citeproc-1.0.4
Fetching: csl-1.4.5.gem (100%)
Successfully installed csl-1.4.5
Fetching: citeproc-ruby-1.1.5.gem (100%)
Successfully installed citeproc-ruby-1.1.5
Fetching: csl-styles-1.0.1.7.gem (100%)
Successfully installed csl-styles-1.0.1.7
Fetching: unicode-0.4.4.2.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing jekyll-scholar:
	ERROR: Failed to build gem native extension.

    /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby extconf.rb
creating Makefile

make "DESTDIR="
make: *** No rule to make target `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/universal-darwin16/ruby/config.h', needed by `unicode.o'.  Stop.
```

Any tips appreciated!

**Comment by edgemaster** [comment]
Looks like the ruby version bundled with xcode doesn't have compilation headers supplied?

Does executing `xcode-select --install` work, as suggested in https://stackoverflow.com/questions/7454361/rubygems-do-not-install-on-os-x-lion

**Comment by heathermiller** [comment]
Thanks for the pointer! Curious though, I had no issues with Ruby compilation headers prior to upgrading to macOS Sierra. (jekyll-scholar is the first gem I've tried since ugprading)

Indeed the above pointer helped me progress, but this isn't the end of the issues. I wonder if this is the result of some reshuffling done in the 10.11->10.12 upgrade process.

```
$ sudo gem install jekyll-scholar
Password:
Building native extensions.  This could take a while...
ERROR:  Error installing jekyll-scholar:
	ERROR: Failed to build gem native extension.

    /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby extconf.rb
creating Makefile

make "DESTDIR="
compiling unicode.c
In file included from unicode.c:13:
In file included from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby.h:33:
In file included from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby/ruby.h:68:
/usr/include/stdio.h:133:19: error: expected ')'
        int     (* _Nullable _close)(void *);
                             ^
/usr/include/stdio.h:133:6: note: to match this '('
        int     (* _Nullable _close)(void *);
                ^
/usr/include/stdio.h:134:19: error: expected ')'
        int     (* _Nullable _read) (void *, char *, int);
                             ^
/usr/include/stdio.h:134:6: note: to match this '('
        int     (* _Nullable _read) (void *, char *, int);
                ^
/usr/include/stdio.h:134:9: error: duplicate member '_Nullable'
        int     (* _Nullable _read) (void *, char *, int);
                   ^
/usr/include/stdio.h:133:9: note: previous declaration is here
        int     (* _Nullable _close)(void *);
                   ^
/usr/include/stdio.h:135:22: error: expected ')'
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                             ^
/usr/include/stdio.h:135:9: note: to match this '('
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                ^
/usr/include/stdio.h:135:12: error: duplicate member '_Nullable'
        fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
                   ^
/usr/include/stdio.h:133:9: note: previous declaration is here
        int     (* _Nullable _close)(void *);
                   ^
/usr/include/stdio.h:136:19: error: expected ')'
        int     (* _Nullable _write)(void *, const char *, int);
                             ^
/usr/include/stdio.h:136:6: note: to match this '('
        int     (* _Nullable _write)(void *, const char *, int);
                ^
/usr/include/stdio.h:136:9: error: duplicate member '_Nullable'
        int     (* _Nullable _write)(void *, const char *, int);
                   ^
/usr/include/stdio.h:133:9: note: previous declaration is here
        int     (* _Nullable _close)(void *);
                   ^
/usr/include/stdio.h:477:25: error: redefinition of parameter '_Nullable'
                 int (* _Nullable)(void *, const char *, int),
                        ^
/usr/include/stdio.h:476:25: note: previous declaration is here
                 int (* _Nullable)(void *, char *, int),
                        ^
/usr/include/stdio.h:478:28: error: redefinition of parameter '_Nullable'
                 fpos_t (* _Nullable)(void *, fpos_t, int),
                           ^
/usr/include/stdio.h:476:25: note: previous declaration is here
                 int (* _Nullable)(void *, char *, int),
                        ^
/usr/include/stdio.h:479:25: error: redefinition of parameter '_Nullable'
                 int (* _Nullable)(void *));
                        ^
/usr/include/stdio.h:476:25: note: previous declaration is here
                 int (* _Nullable)(void *, char *, int),
                        ^
In file included from unicode.c:13:
In file included from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby.h:33:
In file included from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby/ruby.h:70:
In file included from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby/defines.h:29:
/usr/include/stdlib.h:139:38: error: expected ')'
            size_t __width, int (* _Nonnull __compar)(const void *, const void *));
                                            ^
/usr/include/stdlib.h:139:26: note: to match this '('
            size_t __width, int (* _Nonnull __compar)(const void *, const void *));
                                ^
/usr/include/stdlib.h:158:22: error: expected ')'
            int (* _Nonnull __compar)(const void *, const void *));
                            ^
/usr/include/stdlib.h:158:10: note: to match this '('
            int (* _Nonnull __compar)(const void *, const void *));
                ^
/usr/include/stdlib.h:270:38: error: expected ')'
            size_t __width, int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                                            ^
/usr/include/stdlib.h:270:26: note: to match this '('
            size_t __width, int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                                ^
/usr/include/stdlib.h:294:22: error: expected ')'
            int (* _Nonnull __compar)(const void *, const void *));
                            ^
/usr/include/stdlib.h:294:10: note: to match this '('
            int (* _Nonnull __compar)(const void *, const void *));
                ^
/usr/include/stdlib.h:297:22: error: expected ')'
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                            ^
/usr/include/stdlib.h:297:10: note: to match this '('
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                ^
/usr/include/stdlib.h:300:22: error: expected ')'
            int (* _Nonnull __compar)(const void *, const void *));
                            ^
/usr/include/stdlib.h:300:10: note: to match this '('
            int (* _Nonnull __compar)(const void *, const void *));
                ^
/usr/include/stdlib.h:303:22: error: expected ')'
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                            ^
/usr/include/stdlib.h:303:10: note: to match this '('
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                ^
/usr/include/stdlib.h:306:22: error: expected ')'
            int (* _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                            ^
/usr/include/stdlib.h:306:10: note: to match this '('
            int (* _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                ^
/usr/include/stdlib.h:309:22: error: expected ')'
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                            ^
/usr/include/stdlib.h:309:10: note: to match this '('
            int (^ _Nonnull __compar)(const void *, const void *)) __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [unicode.o] Error 1


Gem files will remain installed in /Library/Ruby/Gems/2.0.0/gems/unicode-0.4.4.2 for inspection.
Results logged to /Library/Ruby/Gems/2.0.0/gems/unicode-0.4.4.2/ext/unicode/gem_make.out
```

**Comment by edgemaster** [comment]
That's weird. Whatever c compiler it's using doesn't like some of the syntax of your system's installed header files by the look of it. Is it trying to use a different compiler from the one bundled with xcode? Is xcode up to date and have C/C++ compiler packages installed? (I'm not sure if this is a thing?)

Unfortunately, I'm out of pointers as I'm not a Mac user (but have had limited experience trying to get colleagues' Macs to work, I think I resorted to installing Ruby through RVM and Brew for them...)
The Unicode gem codebase hasn't really changed in years, and should still compile in a correctly configured build environment though. =/

**Comment by edgemaster** [comment]
Potentially related issue here: https://github.com/JuliaLang/julia/issues/18635

**Comment by heathermiller** [comment]
Thanks for your input! So yep, I realized this was an issue with the compilers etc bundled with XCode. 

Poking around a bit, I found that updating to macOS Sierra somehow messed up XCode in general. Updating XCode 8 fixed everything. So this is now resolved! It was some funky issue with XCode upon updating to macOS Sierra, not jekyll-scholar. Sorry for the noise and thanks for your patience/pointers!
